src/utils.c

```diff
@@ -307,6 +307,31 @@ pu_bootpart_enable(const gchar *device,
     return pu_spawn_command_line_sync(cmd, error);
 }
 
+gboolean
+pu_part_set_partuuid(const gchar *device,
+                     guint index,
+                     const gchar *partuuid,
+                     GError **error)
+{
+    g_autofree gchar *cmd = NULL;
+
+    g_return_val_if_fail(g_strcmp0(device, "") > 0, FALSE);
+    g_return_val_if_fail(index > 0, FALSE);
+    g_return_val_if_fail(g_strcmp0(partuuid, "") > 0, FALSE);
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    cmd = g_strdup_printf("sfdisk --part-uuid %s %d %s",
+                          device, index, partuuid);
+
+    if (!pu_spawn_command_line_sync(cmd, error)) {
+        g_prefix_error(error, "Failed setting PARTUUID on %s partition %d: ",
```

Comment:
How about directly printing the device of the partition, like `/dev/mmcblk1p2`? You could use `pu_device_get_partition_path()` to construct the path.

Reply:
I would like to leave it as is, because using `pu_device_get_partition_path()` requires a new empty `GError`.